org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/SignatureHelpHandler.java

```diff
@@ -198,6 +205,26 @@ private boolean isSameParameters(IMethod m, SignatureHelp help, SignatureHelpReq
 		return false;
 	}
 
+	private static boolean isSameParameters(IMethod method1, IMethod method2, CompletionProposal typeProposal) {
+		if (method1 == null || method2 == null) {
+			return false;
+		}
+		String[] params1 = method1.getParameterTypes();
+		String[] params2 = method2.getParameterTypes();
+		if (params2.length == params1.length - 1) {
+			for (int i = 0; i < params2.length; i++) {
+				String t1 = Signature.getSimpleName(Signature.toString(params2[i]));
+				String t2 = Signature.getSimpleName(Signature.toString(params1[i]));
+				if (!t1.equals(t2)) {
+					return false;
+				}
+			}
+		}
+		String param = params1[params1.length - 1];
+		String typeSignature = new String(SignatureUtil.fix83600(typeProposal.getSignature()));
+		return param.equals(typeSignature);

```

Comment:
Isn't this only handling the case of when the cursor is over the last parameter of some matching method signature ?

What about cases like `new StringCharacterIterator("", 0, | , 2);` ( | is the cursor). It seems to get this wrong. I'd actually be fine with not dealing with this for now and addressing them in a later PR.

Reply:
Could you, please, create a new issue?